ServerCore/Helpers/TeamHelper.cs

```diff
@@ -191,7 +249,7 @@ string TruncateName(string name)
 
             string truncatedName = TruncateName(name);
 
-            List<string> existingNames = await (from t in context.Teams where t.EventID == eventObj.ID select t.Name).ToListAsync();
+            List<string> existingNames = (from t in context.Teams where t.EventID == eventId select t.Name).ToList();
```

Comment:
As you spotted below, this was to resolve the validation attribute deadlock issue. It shouldn't have a significant performance impact since it only happens when creating a team